### PR TITLE
NAPSU-MQ inference configuration

### PR DESCRIPTION
**Pull request by oraisa**
Allow changing the inference parameters for NAPSU-MQ. Also add an option to return the `InferenceData` object from NAPSU-MQ that allows inspecting MCMC diagnostics with arviz.

### CODE REVIEW

**Comment by oraisa** [comment]
@lumip The way I used to configure NAPSU-MQ inference is very verbose when changing just one parameter, but it makes the signature of `NapsuMQModel.fit` simpler, as the config is just one argument. Do you think it is too verbose, in which case I could move some parts of the config to be method arguments.

Example from the notebook:
```python
model = NapsuMQModel(required_marginals=required_marginals)
inference_config = NapsuMQInferenceConfig(
    mcmc_config=NapsuMQMCMCConfig(
        num_samples=1000
    )
)
result = model.fit(
    data=orig_df,
    rng=inference_rng,
    epsilon=1,
    delta=(n ** (-2)),
    inference_config=inference_config,
)
```

**Review by lumip — COMMENTED** [review]
I think overall these are good changes. I've added some thoughts about possible improvements in comments directly at the relevant code.

File: twinify/napsu_mq/napsu_mq.py [hunk 1]
```diff
@@ -32,49 +33,112 @@
...
-    @property
-    def uses_laplace_approximation(self) -> bool:
-        return self._use_laplace_approximation
 
     def fit(self, data: pd.DataFrame, rng: d3p.random.PRNGState, epsilon: float, delta: float,
-            query_sets: Optional[Iterable] = None, **kwargs) -> 'NapsuMQResult':
+            query_sets: Optional[Iterable] = None, 
+            inference_config: NapsuMQInferenceConfig = NapsuMQInferenceConfig(),
+            show_progress: bool = True,
+            return_diagnostics: bool = False) -> 'NapsuMQResult':
```
[the remark this answers is not in the document]
Comment (oraisa) [reply twinify/napsu_mq/napsu_mq.py:160]: I don't think using `**kwargs` to just absorb unknown arguments is a good idea, as that can silently ignore errors, for example misspelled argument names. 

The config has a default value, so users can change from DPVI to NAPSU-MQ with the same `fit` call if they want to use the default config. If they are using a custom NAPSU-MQ config, and want to change to DPVI, they should be required to change the `fit` call, as the NAPSU-MQ config won't do anything with DPVI.

I changed to `show_progress`argument to `silent` for compatibility with DPVI.

[the remark this answers is not in the document]
Comment (lumip) [reply twinify/napsu_mq/napsu_mq.py:160]: I can get behind not using `**kwargs` but then we need to sort out the differences in API between the two methods. With the changes here, NAPSU-MQ would have the following extraneous arguments that are not present for DPVI (or the base class `InferenceModel`):
- query_sets
- inference_config
- return_diagnostics

We could make `return_diagnostics` a common argument, that could be handy for DPVI as well - alternatively, we could maybe include the diagnostics as part of the result object.

We could do similarly for `inference_config`, but I prefer making it an argument for `NapsuMQModel.__init__`, i.e., making a `NapsuMQModel` instance completely encapsulate all details of its inference process (`DPVIModel` already mostly functions that way).

What is actually the intended difference between `query_sets` here and `required_marginals` passed in the `__init__`? Could the be unified or both lifted into `__init__`.

Finally, I have reverted your change for `show_progress` and adapted `DPVIModel` accordingly (also removed the `verbose` argument there) - I think `show_progress` is the better/more descriptive argument name here.

[the remark this answers is not in the document]
Comment (oraisa) [reply twinify/napsu_mq/napsu_mq.py:165]: The difference between `query_sets`and `required_marginals` is that one chooses the queries explicitly and the other specifies queries that are always included when other queries are selected automatically. I renamed them to make this clear, and put both to `__init__`.

[the remark this answers is not in the document]
Comment (oraisa) [reply twinify/napsu_mq/napsu_mq.py:167]: I also moved `inference_config`to `__init__`.

**Review by lumip — APPROVED** [review]
This now looks good to me, thanks. I have made some small additional changes to `InferenceModel`  and `DPVIModel` so that all `fit` function now share the same arguments.

**Pull request merged**